src/debugger/VariableManager.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { VariableContainer, Variable } from "./types.js";
+import { ReferenceMap } from "./ReferenceMap.js";
+import { LexicalEnvironment, DeclarativeEnvironmentRecord } from "./../environment.js";
+import { Value, ConcreteValue, PrimitiveValue } from "./../values/index.js";
+
+// This class manages the handling of variable requests in the debugger
+// The DebugProtocol specifies collections of variables are to be fetched using a
+// unique reference ID called a variablesReference. This class can generate new
+// variablesReferences to pass to the UI and then perform lookups for those
+// variablesReferences when they are requested.
+export class VariableManager {
+  constructor() {
+    this._containerCache = new Map();
+    this._referenceMap = new ReferenceMap();
+  }
+  // cache for created references
+  _containerCache: Map<VariableContainer, number>;
+  // map for looking up references
+  _referenceMap: ReferenceMap<VariableContainer>;
+
+  // Given a container, either returns a cached reference for that container if
+  // it exists or return a new reference
+  getReferenceForValue(value: VariableContainer): number {
+    let cachedRef = this._containerCache.get(value);
+    if (cachedRef) {
```

Comment:
Use explicit check here: cachedRef != null. Because JS will treat number 0 as falseness. (I know you never use 0 as a valid reference here, but this falseness checking a potential bug farm we should avoid).